star_realms/src/game/effects/actions.rs

```diff
@@ -142,9 +137,42 @@ pub fn get_action(name: &String) -> Option<Action> {
         }
     }
     match name.as_str() {
+        "goods" => Some(
+            ActionCreator {
+                meta: ActionConfig {
+                    pre_config_meta: Some(
+                        PreConfigMeta::all_required(
+                            vec![
+                                ("combat", PreConfigType::Nat),
+                                ("trade", PreConfigType::Nat),
+                                ("authority", PreConfigType::Nat)
+                            ]
+                        )
+                    )
+                },
+                action: Box::new(|cfg: PreConfig| {
+                    let authority = cfg.get_nat("authority") as Authority;
+                    let trade = cfg.get_nat("trade") as Coin;
+                    let combat = cfg.get_nat("combat") as Combat;
```

Comment:
not poggers, can produce runtime panics at really bad times but what am I supposed to do

Reply:
write good code? bruh